### PR TITLE
Add rotate screen support. #14 and #15

### PR DESCRIPTION
**Pull request by pedrovgs**
To be able to implement the feature requested at #14 and the example requested at #15 I've implemented the screen rotation support. This could be implemented out of the library but it was quite straightforward to implemented it inside Dexter, so I decided to add this feature to the project instead of add it to the example project.

To be able to support this feature the only thing our users has to do is to call a method in the ``onCreate``   Activity method where the library is used. I've updated the ``README.md`` with some documentation.

Please @greysonp check if this feature is enough or if you need something more specific.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -127,6 +127,16 @@ MultiplePermissionsListener dialogMultiplePermissionsListener = /*...*/;
 Dexter.checkPermissions(new CompositePermissionListener(snackbarMultiplePermissionsListener, dialogMultiplePermissionsListener, /*...*/), Manifest.permission.CAMERA, Manifest.permission.RECORD_AUDIO);
 ```
 
+** If your application has to support configuration changes based on screen rotation remember to add a call to ``Dexter`` in your Activity ``onCreate`` method as follows:**
```
Comment (Serchinastico) [inline README.md]: Add a header at the same level as Single permission/Multiple permissions to explain this as it's something that will be useful for both. Something like **Handling Activity rotation**.

Comment (pedrovgs) [inline README.md]: Done :)

File: dexter/src/main/java/com/karumi/dexter/DexterInstance.java [hunk 2]
```diff
@@ -89,11 +90,22 @@ void checkPermissions(MultiplePermissionsListener listener, Collection<String> p
...
 
   /**
+   * Check if there are some permissions pending to be confirmed by the user and restarts the request
+   * for permission process.
+   */
+  void checkPendingPermissions() {
+    boolean shouldContinueRequestingPendingPermissions =
+        !pendingPermissions.isEmpty() && !rationaleShown.get();
+    if (shouldContinueRequestingPendingPermissions) {
+      startTransparentActivity();
```
Comment (Serchinastico) [inline dexter/src/main/java/com/karumi/dexter/DexterInstance.java]: Wouldn't this potentially leak activities? the old activity hasn't been destroyed yet we are creating another one and leaving the old one alive. Given this class is entirely responsible for the transparent activity lifecycle I'd make sure we are only handling one at a time, either closing the old one and relaunching a new one or reusing the original.

**Comment by flipper83** [comment]
:+1:

**Comment by pedrovgs** [comment]
With the help of @Serchinastico we've added some changes to avoid memory issues with the transparent activity recreated when the device was rotated. Thanks for everyone involved. We are going to wait for @greysonp before to merge this PR. Maybe he has something to add.

**Comment by Serchinastico** [comment]
:+1:

**Comment by greysonp** [comment]
Hey I probably won't be available until Sunday-ish because of the holiday. I'll check it out when I get a chance!

**Comment by pedrovgs** [comment]
@Serchinastico shouldn't we assign the listener to null after the request permission process?

**Comment by pedrovgs** [comment]
Done :) I've replaced the listener with an empty implementation.

**Comment by Serchinastico** [comment]
:+1:

File: dexter/src/main/java/com/karumi/dexter/DexterInstance.java [hunk 3]
```diff
@@ -86,15 +91,38 @@ void checkPermissions(MultiplePermissionsListener listener, Collection<String> p
...
+  }
+
+  /**
+   * Check if there are some permissions pending to be confirmed by the user and restarts the
+   * request for permission process.
+   */
+  void checkPendingPermissions(MultiplePermissionsListener listener) {
+    boolean shouldContinueRequestingPendingPermissions =
+        !pendingPermissions.isEmpty() && !rationaleAccepted.get();
+    this.listener = listener;
```
Comment (greysonp) [inline dexter/src/main/java/com/karumi/dexter/DexterInstance.java]: Should we always be assigning the listener, even if ```!shouldContinueRequestingPendingPermissions```?

**Comment by greysonp** [comment]
I don't know the finer details of the system, but the end interface provided to the user looks good. My only bit of confusion lies with the effect of ```Dexter.checkPendingPermissions()```. It looks like the listener passed in will always be assigned, even if there's no pending request in progress. That isn't implied by the docs nor the name of the method. Instead, the name just make you think this is some sort of check that will return a boolean.

If the goal is to both check if a request is in progress *and* assign the listener, then the name and docs should reflect that.

Suggestions:
* Remove the listener if there's no pending permission requests and rename the method to ```continuePendingRequestsIfPossible()``` or something to that effect
* Keep the assignment and rename the method ```assignListenerAndContinuePendingRequests()``` or something to that effect

**Comment by Serchinastico** [comment]
Hi @greysonp 

The proposal to replace ``checkWhatever`` with ``continueWhatever`` makes sense to me, but the ``assignListener`` part looks like exposing an implementation detail that shouldn't be explicitly declared in the public API. I'm ok including it in the docs though.

File: dexter/src/main/java/com/karumi/dexter/DexterInstance.java (hunk 3 above)
Comment (Serchinastico) [inline dexter/src/main/java/com/karumi/dexter/DexterInstance.java]: Probably not, we are keeping a ref to something we are not going to use (and clean) so I'm moving it to the if block. Thanks!

**Comment by greysonp** [comment]
FWIW people need to know if they're about to hand something off to a library that's going to be held statically in memory (like this listener). Lots of potential for memory leaks here. That's the only reason I wanted to make it clear that this listener is going to be used no matter what.

It's probably fine here though since the act of giving a listener to a static method should be enough to tip someone off, but it's definitely something people need to be aware of in general with a library like this.

**Comment by Serchinastico** [comment]
Hi @greysonp , I have created a PR to fix the issue you pointed out in ``continuePendingRequestsIfPossible`` [https://github.com/Karumi/Dexter/pull/28] I'm sorry for the misunderstanding.

Regarding the static memory, I'd say that the fact Dexter is being called using static methods can give you the idea that we are holding static references (how else would it be possible to hold state between calls?). Now that the listener reference is cleaned up once the request permission process finishes (and doesn't keep it if it's not needed) I don't see the point of declaring that we are going to hold a reference to it.

**Pull request merged**